js/app/controllers/calcontroller.js

```diff
@@ -140,15 +140,18 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 				$scope.$apply();
 			});
 		} else {
-			$scope.calendarsPromise = CalendarService.getPublicCalendar(constants.publicSharingToken).then(function(calendar) {
-				$scope.calendars = [calendar];
-				is.loading = false;
-				// TODO - scope.apply should not be necessary here
-				$scope.$apply();
+			constants.publicSharingToken.split(".").forEach(token=>
+			{
+				$scope.calendarsPromise = CalendarService.getPublicCalendar(token).then(function(calendar) {
+					$scope.calendars = [calendar];
```

Comment:
If I'm not wrong, this will overwrite `$scope.calendars` with an array of the current calendar. You need to push it instead, like `$scope.calendars.push(calendar);`.